joomla/joomla-cms
# [5.3] Deprecate HTMLHelper::script(), HTMLHelper::stylesheet(), extend lifetime for Document style/script methods

Pull Request for Issue # .

### Summary of Changes

Deprecate `HTMLHelper::script()`, `HTMLHelper::stylesheet()`.
Extend lifetime for `Document` style/script methods, because they cannot be removed before `HTMLHelper::script()`, `HTMLHelper::stylesheet()`.

Add a new method `HTMLHelper::mediaPath()`, because we still need to lookup for overides. 
Kind of temporary solution, until someone will come up with improvements to our media override API.


### Testing Instructions

Apply patch. Make sure all works as before.

Add `media/templates/administrator/atum/css/user.css` to atum template, and make sure it still works.


### Actual result BEFORE applying this Pull Request

Works

### Expected result AFTER applying this Pull Request

Works


### Link to documentations
Please select:
- [x] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/379
